### PR TITLE
EventHub: updated help for the parameter --partition-count of eventhub

### PR DESCRIPTION
**Pull request by v-Ajnava**
---
* updated help for the parameter --partition-count of eventhub

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [X] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/command_modules/azure-cli-eventhubs/azure/cli/command_modules/eventhubs/_params.py [hunk 1]
```diff
@@ -76,7 +76,7 @@ def load_arguments_eh(self, _):
     for scope in ['eventhubs eventhub update', 'eventhubs eventhub create']:
         with self.argument_context(scope) as c:
             c.argument('message_retention_in_days', options_list=['--message-retention'], type=int, help='Number of days to retain events for this Event Hub, value should be 1 to 7 days')
-            c.argument('partition_count', type=int, help='Number of partitions created for the Event Hub, allowed values are from 1 to 32 partitions.')
+            c.argument('partition_count', type=int, help='Number of partitions created for the Event Hub, allowed values are from 2 to 32 partitions. Default lower limit of 1 is supported with Kafka enabled namespaces. Default upper limit is 32 unless you have custom quota, in which case it would be upper limit of that quota.')
```
Comment (williexu) [inline src/command_modules/azure-cli-eventhubs/azure/cli/command_modules/eventhubs/_params.py]: It sounds from the additional help you added that the allowed values are not restrained from 2-32? At least for some users?

Comment (v-Ajnava) [reply src/command_modules/azure-cli-eventhubs/azure/cli/command_modules/eventhubs/_params.py]: yes, by default the upper limit is of 32, which can be increased by raising the quota request.

Comment (williexu) [reply src/command_modules/azure-cli-eventhubs/azure/cli/command_modules/eventhubs/_params.py]: Right, my point is, the wording: "allowed values are from 2 to 32 partitions" implies a strict boundary, i.e. NO ONE is able to specify `--partition-count 1` or `--partition-count 33`.

The wording could be improved. Recommend:
`By default, allowed values are 2-32. Value of 1 is supported with Kafka enabled namespaces. In presence of a custom quota, the upper limit will match the upper limit of the quota.`

Comment (v-Ajnava) [reply src/command_modules/azure-cli-eventhubs/azure/cli/command_modules/eventhubs/_params.py]: updated with : Number of partitions created for the Event Hub. By default, allowed values are 2-32. Lower value of 1 is supported with Kafka enabled namespaces. In presence of a custom quota, the upper limit will match the upper limit of the quota.

**Comment by troydai** [comment]
@williexu approve and merge when you deem suitable. let's try to merge it in 2.0.43 release.

**Pull request merged**